manifests/init.pp

```diff
@@ -113,92 +45,49 @@
-  Enum['file', 'syslog'] $qpid_router_logging = $foreman_proxy_content::params::qpid_router_logging,
-  Stdlib::Absolutepath $qpid_router_logging_path = $foreman_proxy_content::params::qpid_router_logging_path,
-  Optional[String] $qpid_router_ssl_ciphers = $foreman_proxy_content::params::qpid_router_ssl_ciphers,
-  Optional[Array[String]] $qpid_router_ssl_protocols = $foreman_proxy_content::params::qpid_router_ssl_protocols,
-  Optional[String] $qpid_router_sasl_mech = $foreman_proxy_content::params::qpid_router_sasl_mech,
-  Optional[String] $qpid_router_sasl_username = $foreman_proxy_content::params::qpid_router_sasl_username,
-  Optional[String] $qpid_router_sasl_password = $foreman_proxy_content::params::qpid_router_sasl_password,
-  Boolean $enable_ostree = $foreman_proxy_content::params::enable_ostree,
-  Boolean $enable_yum = $foreman_proxy_content::params::enable_yum,
-  Boolean $enable_file = $foreman_proxy_content::params::enable_file,
-  Boolean $proxy_pulp_isos_to_pulpcore = $foreman_proxy_content::params::proxy_pulp_isos_to_pulpcore,
-  Boolean $proxy_pulp_yum_to_pulpcore = $foreman_proxy_content::params::proxy_pulp_yum_to_pulpcore,
-  Boolean $enable_puppet = $foreman_proxy_content::params::enable_puppet,
-  Boolean $enable_docker = $foreman_proxy_content::params::enable_docker,
-  Boolean $enable_deb = $foreman_proxy_content::params::enable_deb,
-
-  Boolean $manage_broker = $foreman_proxy_content::params::manage_broker,
-
-  Boolean $pulpcore_manage_postgresql = $foreman_proxy_content::params::pulpcore_manage_postgresql,
-  Stdlib::Host $pulpcore_postgresql_host = $foreman_proxy_content::params::pulpcore_postgresql_host,
-  Stdlib::Port $pulpcore_postgresql_port = $foreman_proxy_content::params::pulpcore_postgresql_port,
-  String $pulpcore_postgresql_user = $foreman_proxy_content::params::pulpcore_postgresql_user,
-  String $pulpcore_postgresql_password = $foreman_proxy_content::params::pulpcore_postgresql_password,
-  String $pulpcore_postgresql_db_name = $foreman_proxy_content::params::pulpcore_postgresql_db_name,
-  Boolean $pulpcore_postgresql_ssl = $foreman_proxy_content::params::pulpcore_postgresql_ssl,
-  Boolean $pulpcore_postgresql_ssl_require = $foreman_proxy_content::params::pulpcore_postgresql_ssl_require,
-  Stdlib::Absolutepath $pulpcore_postgresql_ssl_cert = $foreman_proxy_content::params::pulpcore_postgresql_ssl_cert,
-  Stdlib::Absolutepath $pulpcore_postgresql_ssl_key = $foreman_proxy_content::params::pulpcore_postgresql_ssl_key,
-  Stdlib::Absolutepath $pulpcore_postgresql_ssl_root_ca = $foreman_proxy_content::params::pulpcore_postgresql_ssl_root_ca,
+  Stdlib::Fqdn $parent_fqdn = $foreman_proxy_content::params::parent_fqdn, # TODO: extract this from the Foreman URL as passed to foreman_proxy?
+
+  Boolean $puppet = true,
+
+  Boolean $reverse_proxy = false,
+  Stdlib::Port $reverse_proxy_port = 8443,
+
+  Optional[String] $rhsm_hostname = undef,
+  String $rhsm_url = '/rhsm', # TODO: Not really a URL
```

Comment:
Probably shouldn't be configurable to be honest, the API is the API.

Reply:
Having it live in a variable is generally a good thing and I think this still has some room to split functionality in smaller classes. Then you can make it a parameter on the private class API but not expose it in the public class API.

Reply:
I'm good with it being a variable, I meant the perceived intent that a user can and should *change* the value is wrong. This is a hard-coded API in the Katello plugin so having this be changeable by someone using the public interface of this module can result in incorrect behavior.

Reply:
Agreed, but I'm first going to focus on the Pulp parts. It should probably be done as part of https://github.com/theforeman/puppet-foreman_proxy_content/pull/272 since that does introduce the private API I was thinking about.